sgx/core-types/src/key_request.rs

```diff
@@ -324,6 +335,98 @@ impl Display for KeyRequest {
     }
 }
 
+impl FfiWrapper<sgx_key_request_t> for KeyRequest {}
+
+impl Hash for KeyRequest {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        "KeyRequest".hash(state);
+        self.key_name().hash(state);
+        self.key_policy().hash(state);
+        self.security_version().hash(state);
+        self.cpu_security_version().hash(state);
+        self.attribute_mask().hash(state);
+        self.key_id().hash(state);
+        self.misc_mask().hash(state);
+        self.config_security_version().hash(state);
+    }
+}
+
+// TODO: Implement prost::Message
+
+impl Ord for KeyRequest {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.key_name().cmp(&other.key_name()).then(
```

Comment:
Seconding @mfaulk's comment that it might be easier to follow if a tuple is constructed with all these values and then compared.

Reply:
OK, will address